### PR TITLE
mgr/dashboard: n/a entries behind primary snapshot mode

### PR DESCRIPTION
**Pull request by pereman2**
image

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by nizamial09 — APPROVED** [review]
looks good to me! thank you @pereman2

**Comment by nizamial09** [comment]
ccould you attach the tracker here? @pereman2

File: src/pybind/mgr/dashboard/controllers/rbd_mirroring.py [hunk 1]
```diff
@@ -238,6 +238,17 @@ class ReplayingData(NamedTuple):
     entries_behind_primary: Optional[int] = None
 
 
+def _get_mirror_mode(ioctx, image_name):
```
Comment (avanthakkar) [inline src/pybind/mgr/dashboard/controllers/rbd_mirroring.py:241]: Type annotations would be good to have for any new func introduced.

File: src/pybind/mgr/dashboard/controllers/rbd_mirroring.py [hunk 2]
```diff
@@ -300,7 +311,8 @@ def _get_pool_datum(pool_name):
         data['mirror_images'] = sorted([
             dict({
                 'name': image['name'],
-                'description': image['description']
+                'description': image['description'],
+                'mirror_mode': _get_mirror_mode(ioctx, image['name'])
```
Comment (avanthakkar) [inline src/pybind/mgr/dashboard/controllers/rbd_mirroring.py:315]: Can you add some unit test for this. I see there's `RbdMirroringSummaryControllerTest` so probably a test for checking `mirror_mode` field there would be good to have.

**Comment by nizamial09** [comment]
@pereman2 ping^

**Pull request merged**